Offroad-iOS/Offroad-iOS/Presentation/MyPage/CharacterList/ViewController/CharacterDetailViewController.swift

```diff
@@ -11,14 +11,23 @@ final class CharacterDetailViewController: UIViewController {
     
     // MARK: - Properties
     
-    private var imageName: String
+    private let characterId: Int
     
     private let characterDetailView = CharacterDetailView()
+    private var characterDetailList: [CharacterDetailList]?
+    
+    private var characterMainColorCode: String?
+    private var characterSubColorCode: String?
+    
+    private var combinedCharacterMotionList: [(isGained: Bool, character: Any)] = []
+    private var gainedCharacterMotionList: [GainedCharacterMotionList]?
+    private var notGainedCharacterMotionList: [NotGainedCharacterMotionList]?
```

Comment:
P2

위 코멘트와 연결해서 `CharacterMotionList` 구조체로 변경하여 다음과 같이 코드를 바꿔줄 수 있을 것 같아요!
```suggestion
    private var combinedCharacterMotionList: [CharacterMotionList]? //(여기에다 프로퍼티 옵저버를 걸어주면 데이터 처리에도 더 좋을 것 같네용)
    private var gainedCharacterMotionList: [CharacterMotionList]?
    private var notGainedCharacterMotionList: [CharacterMotionList]?
```

Reply:
프로퍼티 옵저버를 거는 것은 적용했지만, `private var combinedCharacterMotionList: [CharacterMotionList]?` 이렇게 설정하게 되면 CharacterMotionList에 isGained 속성이 없기 때문에 추후 notGainedCharacterMotionList의 셀에 불투명한 배경과 자물쇠 이미지를 넣는 로직이 불편해지는 문제가 생깁니다. 따라서 아래와 같이 구현했습니다.
```swift
private var combinedCharacterMotionList: [(isGained: Bool, character: Any)] = [] {
        didSet {
            DispatchQueue.main.async {
                self.characterDetailView.collectionView.reloadData()
            }
        }
    }
```